fetcher/es-fetcher/league.go

```diff
@@ -0,0 +1,70 @@
+package fetcher
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/serhiihuberniuk/bet-predictor/fetcher/es-fetcher/es-models"
+	"github.com/serhiihuberniuk/bet-predictor/models"
+)
+
+func (f *ESFetcher) AllLeaguesList(ctx context.Context) ([]*models.League, error) {
+
+	remoteLeagues := make([]esmodels.League, 0)
+	page := 1
+
+	for {
+		leaguesResponse, err := f.getPageOfLeagues(ctx, page)
+		if err != nil {
+			return nil, fmt.Errorf("error while getting leagues from page %v: %w", page, err)
+		}
+
+		remoteLeagues = append(remoteLeagues, leaguesResponse.Data...)
+
+		if !leaguesResponse.Pagination.HasNextPage {
+			break
+		}
+
+		page++
+	}
+
+	var leagues []*models.League
+	for _, league := range remoteLeagues {
+		leagues = append(leagues, &models.League{
+			Name:    league.Name,
+			Country: league.CountryName,
+		})
```

Comment:
it's not correct to use model 'League', you need to create a new model `LeaguePayload` or something like that. The reason is that you should always try to fill all fields in a model, if you can't - create a new model